DEVELOPERS.md

```diff
@@ -0,0 +1,134 @@
+
+There is a `CopierFactoryManager` which determines which type of copier will be used. The `DefaultCopierFactoryManager` is an implementation of this, and has a list of `CopierFactories` auto-wired into it. Spring will find all beans which implement `CopierFactory` and pass these on to the constructor for the `DefaultCopierFactoryManager`. 
+
+There is an optional copier option available to set which `CopierFactory` to use, if this value is set this copier factory class will be used. If this value is not set the `DefaultCopierFactoryManager` will check all `CopierFactories` in the list and return the first which supports replication between the SourceLocation and ReplicaLocation provided. 
+
+There is an order of precedence, which means the `CopierFactories` will be checked in the following order to see if the replication is supported:
+* `S3S3Copier`,
+* `S3MapreduceCpCopier`, 
+* and then falls down to `DistCpCopier` if the above factories don't support the replication.
+
+The copiers which use S3 will create clients that allow access and give permissions to perform actions on S3 buckets. In some cases an IAM role is needed, if data is being transferred across S3 accounts. 
+
+### Types of copier
+**S3S3Copier**
+
+*Replication: S3 → S3* 
+
+This copier uses two `AwsS3Clients` - a source client and a replica client. There is an `AwsS3ClientFactory` which will create clients to perform actions on S3 buckets. 
+
+One of these client factories is `JceksAmazonS3ClientFactory`, which creates a client with the necessary credentials required. It does this using a credential provider chain, which will create (as the name states) a chain of credential providers which will be tried in order, until one is successful. One of the credentials in this chain is the `AssumeRoleCredentialProvider` which uses a role provided in the copier options to be able to replicate across S3 accounts.
+
+The replication is handled by a `TransferManager` which uses the target S3 client and the `S3S3CopierOptions`. The `TransferManager` will be given the the source client to replicate from. 
+
+The `S3S3CopierOptions` will take the `CopierOptions` provided and change them into more specific s3 options. For example it will have the options `s3-server-side-encryption` and `assume-role`, which are specific to S3 clients and won't be used by the other copiers. 
+
+*Cross-account Replication*
+
+As mentioned above, if you want to replicate from one S3 account to another S3 account you will need to make use of `roles`. 
+In the config file for the replication an `assume-role` will need to be included which will be used by Circus Train to perform the replication. In order for this to work this role will need to have the following permissions:
+* Read access to the source account, 
+* Read and write access to the target account. 
+
+Its important that the role has read access to the source so the data can be read before it is replicated to the target. 
```

Comment:
I think we can remove this line, IMHO it's clear that if you can't read from the source then you can't do much!